libraries/chain/asset_evaluator.cpp

```diff
@@ -404,24 +405,28 @@ void_result asset_update_bitasset_evaluator::do_apply(const asset_update_bitasse
    // We also need to update median feeds, as well as check call orders
    // if the feed_lifetime_sec changed (after Hardfork core 890)
    bool after_hardfork_890 = false;
-   if (db().head_block_time() >= HARDFORK_CORE_890_TIME
+   if (db_conn.get_dynamic_global_properties().next_maintenance_time > HARDFORK_CORE_890_TIME
          && o.new_options.feed_lifetime_sec != bitasset_to_update->options.feed_lifetime_sec)
    {
       should_update_feeds = true;
       after_hardfork_890 = true;
    }
 
-   db().modify(*bitasset_to_update, [&](asset_bitasset_data_object& b) {
+   db().modify(*bitasset_to_update, [ &o, should_update_feeds, after_hardfork_890, &db_conn ](asset_bitasset_data_object& b) {
       b.options = o.new_options;
 
       if( should_update_feeds )
-         b.update_median_feeds(db().head_block_time());
+      {
+         auto old_feed_price = b.current_feed.settlement_price;
+         b.update_median_feeds(db_conn.head_block_time());
+         if (after_hardfork_890
```

Comment:
Since `after_hardfork_890` was not initialized with hard fork time check only, the logic here is incorrect. Actually we need to call `check_call_order` no matter if `feed_lifetime_sec` is changed.